### PR TITLE
Pinned TF version and limited providers versions

### LINKED ISSUE #118: Required providers are not configured
Hello, in the **main.tf** file seems to be a deprecated option on **subnetwork** config called "enable_flow_logs". It seems to be replaced for log_config.

When I tried to execute a terraform plan using this module, it is returning this message:

**_Error: "enable_flow_logs": [REMOVED] This field is being removed in favor of log_config. If log_config is present, flow logs are enabled. Please remove this field_**

If is possible, could you remove this field from main.tf file?

Thanks in advance.

Best regards.

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Fix #118

### CODE REVIEW

File: examples/delete_default_gateway_routes/main.tf [hunk 1]
```diff
@@ -14,14 +14,6 @@
  * limitations under the License.
  */
 
-provider "google" {
```
Comment (morgante) [inline examples/delete_default_gateway_routes/main.tf]: @paulpalamarchuk Why are you removing these? We specifically do want them.

Comment (paulpalamarchuk) [reply examples/delete_default_gateway_routes/main.tf]: Moved to the `versions.tf`.

**Comment by morgante** [comment]
@paulpalamarchuk Please rebase.

**Comment by paulpalamarchuk** [comment]
@morgante Do we need exact "2.19.x" version of the google provider? Or I can limit it to `3> && >=2.19`?

File: examples/delete_default_gateway_routes/main.tf (hunk 1 above)
Comment (morgante) [reply examples/delete_default_gateway_routes/main.tf]: I have no idea why you removed these. They're exactly right as they are. Please do not change anything about this line, this is exactly how provider versions should be declared in examples.

File: test/setup/versions.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
```
Comment (morgante) [inline test/setup/versions.tf]: Do not change this pinning approach. It is how we should do pinning within `test/setup`.

**Comment by morgante** [comment]
Superseded by #130.

**Pull request closed without merging**